libr/bp/bp.c

```diff
@@ -409,3 +420,22 @@ R_API int r_bp_size(RBreakpoint *bp) {
 	}
 	return bpsize;
 }
+
+// Check if the breakpoint is inside a valid map
+R_API bool r_bp_is_valid(RBreakpoint *bp, RBreakpointItem *b) {
+	RDebugMap *map;
+	RListIter *iter;
+	RCore *core = (RCore*)bp->corebind.core;
+	if (!r_config_get_i (core->config, "dbg.bpinmaps")) {
+		return true;
+	}
+
+	r_debug_map_sync (core->dbg);
```

Comment:
use corebind api like bp->corebind.mapSync() but this is not implemented . see libr/include/r_bind.h and r_core_bind function

Reply:
but.. is this call really needed? cant we just call it from rdebug? also, this is a very expensive call to be called in isvalid()

Reply:
or maybe corebind.isMapped(addr)

Reply:
Added getDebugMaps since isMapped doesn't check file permissions and also calls r_debug_map_sync for each addr.

> but.. is this call really needed? cant we just call it from rdebug?

Limited it to once per restore (if bpinmaps is enabled).